### ISSUE TITLE
Add Enum initializers from rawValue

### ISSUE THREAD

**Issue by angelolloqui** [angelolloqui/SwiftKotlin#94]
In Swift, when you have an enum like:

```
public enum PaymentMethodType: String {
    case direct = "DIRECT",  creditCard = "CREDIT_CARD"
}
```

The you can instantiate the enum from a rawValue like:
`let type = PaymentMethodType(rawValue: "DIRECT")`

In Kotlin, the equivalent would be:

```
enum class PaymentMethodType constructor(var rawValue: String) {
    direct("DIRECT"),  creditCard("CREDIT_CARD");
}
```

But that does not provide the Swift constructor from `rawValue`.

An option, would be to create an additional method on the Kotlin code like:

```
enum class PaymentMethodType constructor(var rawValue: String) {
    direct("DIRECT"),  creditCard("CREDIT_CARD");

    companion object {
        fun from(rawValue: String): PaymentMethodType? =
                values().firstOrNull { it.rawValue == rawValue }
    }
}
```

But this also affects the way the caller needs to invoke the instantiation to:

`val type = PaymentMethodType.from(rawValue = "DIRECT")`

**Comment by angelolloqui** [comment]
For reference, there is also a `EnumClass.valueOf(value: String): EnumClass` method in Kotlin out of the box, but that one throws an exception if the value is not found, while the Swift version returns an optional

**Comment by angelolloqui** [comment]
Wait, we should try this approach:

```
enum class PaymentMethodType constructor(var rawValue: String) {
    direct("DIRECT"),  creditCard("CREDIT_CARD");

    companion object {
         operator fun invoke(rawValue: String) = PaymentMethodType.values().firstOrNull { it.rawValue == rawValue }
    }
}
```

Because, apparently that works as well, and allows to keep the same caller code:

`val method = PaymentMethodType(rawValue = "DIRECT")`

**Comment by torlangballe** [comment]
Great!

Strange way that constructor is made :-)

**Comment by torlangballe** [comment]
Do you want me to add this in a new branch, or will you do it?

**Comment by angelolloqui** [comment]
I think you already got some stuff working for this, didn't you? if so, please go ahead. Otherwise, I will try to find some time, but as usual, I can not promise on dates.

**Comment by angelolloqui** [comment]
Merged with #97
